waltid-services/waltid-wallet-api/src/main/kotlin/id/walt/webwallet/service/SSIKit2WalletService.kt

```diff
@@ -513,25 +517,65 @@ class SSIKit2WalletService(
     }
 
     override suspend fun deleteKey(alias: String): Boolean = runCatching {
-        KeysService.get(walletId, alias)?.let { Json.parseToJsonElement(it.document) }?.run {
+        val key = KeysService.get(walletId, alias)
```

Comment:
if I may suggest a refactoring for `deleteKey` and `removeKey` like following:
- pull the common logic in a separate method - performKeyDelete
  - here I used the `getKey` method already available in the class to retrieve the key
  - then I perform the actions according to inputs
  - I check the operation result and throw explicitly (when remote delete fails)
```kotlin
private suspend fun performKeyDelete(alias: String, isTotalDelete: Boolean) = runCatching {
    val key = getKey(alias)
    val canDeleteFromStorage = isTotalDelete && key.deleteKey() || !isTotalDelete
    val opSucceed = canDeleteFromStorage && KeysService.delete(walletId, alias)
    if (isTotalDelete && !opSucceed) throw WebException(
        HttpStatusCode.BadRequest,
        "Failed to delete remote key : $alias"
    )
    Pair(opSucceed, key)
}.onSuccess {
    eventUseCase.log(
        action = EventType.Key.Delete,//todo: add delete type - remote vs. ref
        originator = "wallet",
        tenant = tenant,
        accountId = accountId,
        walletId = walletId,
        data = eventUseCase.keyEventData(
            id = alias,
            algorithm = it.second.keyType.name,
            kmsType = EventDataNotAvailable
        )
    )
}.onFailure {
    val errorMessage = "Failed to delete key: ${it.message}"
    logger.error(it) { errorMessage }
    throw WebException(HttpStatusCode.BadRequest, errorMessage)
}
```
- then call it in the deleteKey and removeKey methods:
```kotlin
override suspend fun deleteKey(alias: String): Boolean =
    performKeyDelete(alias = alias, isTotalDelete = true).getOrThrow().first


override suspend fun removeKey(alias: String): Boolean =
    performKeyDelete(alias = alias, isTotalDelete = false).getOrThrow().first
```